src/location-rules.php

```diff
@@ -0,0 +1,977 @@
+
+	/**
+	 * The field groups to map to location rules
+	 *
+	 * @var array|mixed
+	 */
+	public $acf_field_groups = [];
+
+	/**
+	 * LocationRules constructor.
+	 *
+	 * @param array $acf_field_groups
+	 */
+	public function __construct( $acf_field_groups = [] ) {
+		$this->acf_field_groups = isset( $acf_field_groups ) && ! empty( $acf_field_groups ) ? $acf_field_groups : acf_get_field_groups();
+	}
+
+	/**
+	 * Given a field name, formats it for GraphQL
+	 *
+	 * @param string $field_name The field name to format
+	 *
+	 * @return string
+	 */
+	public function format_field_name( string $field_name ) {
+
+		$replaced = preg_replace( '[^a-zA-Z0-9 -]', '_', $field_name );
+
+		// If any values were replaced, use the replaced string as the new field name
+		if ( ! empty( $replaced ) ) {
+			$field_name = $replaced;
+		}
+
+		$field_name = lcfirst( $field_name );
+		$field_name = lcfirst( str_replace( '-', ' ', ucwords( $field_name, '_' ) ) );
+		$field_name = lcfirst( str_replace( ' ', '', ucwords( $field_name, ' ' ) ) );
+
+		return $field_name;
+	}
+
+	/**
+	 * Given a type name, formats it for GraphQL
+	 *
+	 * @param string $type_name The type name to format
+	 *
+	 * @return string
+	 */
+	public function format_type_name( string $type_name ) {
+		return ucfirst( $this->format_field_name( $type_name ) );
+	}
+
+	/**
+	 * Given the name of a GraphqL Field Group and the name of a GraphQL Type, this sets the
+	 * field group to show in that Type
+	 *
+	 * @param string $field_group_name  The name of the ACF Field Group
+	 * @param string $graphql_type_name The name of the GraphQL Type
+	 */
+	public function set_graphql_type( string $field_group_name, string $graphql_type_name ) {
+		$this->mapped_field_groups[ Utils::format_field_name( $field_group_name ) ][] = $this->format_type_name( $graphql_type_name );
+	}
+
+	/**
+	 * Given the name of a GraphqL Field Group and the name of a GraphQL Type, this unsets the
+	 * GraphQL Type for the field group
+	 *
+	 * @param string $field_group_name  The name of the ACF Field Group
+	 * @param string $graphql_type_name The name of the GraphQL Type
+	 */
+	public function unset_graphql_type( string $field_group_name, string $graphql_type_name ) {
+		$this->unset_types[ $this->format_field_name( $field_group_name ) ][] = $this->format_type_name( $graphql_type_name );
+	}
+
+	/**
+	 * Get the rules
+	 *
+	 * @return array
+	 */
+	public function get_rules() {
+
+		$mapped_field_groups = isset( $this->mapped_field_groups ) && ! empty( $this->mapped_field_groups ) ? $this->mapped_field_groups : [];
+
+		if ( empty( $mapped_field_groups ) ) {
+			return [];
+		}
+
+
+		if ( empty( $this->unset_types ) ) {
+			return $mapped_field_groups;
+		}
+
+		/**
+		 * Remove any Types that were flagged to unset
+		 */
+		foreach ( $this->unset_types as $field_group => $types ) {
+			if ( ! empty( $types ) ) {
+				foreach ( $types as $type ) {
+					if ( isset( $this->mapped_field_groups[ $field_group ] ) ) {
```

Comment:
Looks like this if isset `mapped_field_groups[ $field_group ]` could be moved outside the foreach ($types).

Reply:
And should the $this->mapped_field_groups be $mapped_field_groups for this is isset check?

Reply:
@markkelnar super helpful feedback! 

I just pushed up some changes to simplify this and stepping through caused me to find a small bug in the JS as well! 

Super cool. Thanks for taking the time to go through this hot mess of a PR 😄